tests/scripts/auto-grow-storage.sh

```diff
@@ -0,0 +1,143 @@
+      unitsize="Mi"
+   elif [[ "$currentsize" == *"Gi" ]]
+   then
+      rawsize=$(echo $currentsize | sed -e 's/Gi//g')
+      unitsize="Gi"
+   elif [[ "$currentsize" == *"Ti" ]]
+   then
+      rawsize=$(echo $currentsize | sed -e 's/Ti//g')
+      unitsize="Ti"
+   else
+      echo "Unknown unit this PVC: ${currentsize}"
+   fi
+   newsize=$(echo "${rawsize}+(${rawsize} * ${growRate})/100" | bc | cut -f1 -d'.')
+   if [ "${newsize}" = "${rawsize}" ]
+   then
+      newsize=$(( rawsize + 1 ))
+      echo "New adjusted calculated size for the PVC is ${newsize}${unitsize}"
+   else
+      echo "New calculated size for the PVC is ${newsize}${unitsize}"
+   fi
+   if [ "${newsize}" -gt "${maxSize}" ]
+   then
+      newsize=${maxSize}
+      echo "Disk has reached it's MAX capacity, add a new disk to it"
+   fi
+   result=$(oc patch pvc ${pvc} -n ${ns} --type json --patch  "[{ "op": "replace", "path": "/spec/resources/requests/storage", "value": "${newsize}${unitsize}" }]")
+   echo ${result}
+}
+
+function growHorizontally() {
+   count=$1
+   pvc=$2
+   ns=$3
+   maxCount=$4 
+   deviceSetName=$(oc get pvc  ${pvc}  -n ${ns}  -o json | jq -r '.metadata.labels."ceph.rook.io/DeviceSet"')
+   cluster=""
+   deviceSets=""
+   currentCount=0
+   clusterCount=0
+   deviceSetCount=0
+   while [ -z "$cluster" ]
+   do
+      cluster=$(oc get CephCluster ${ns} -o json | jq -r ".items["$clusterCount"]")
+      while  [ -z "$deviceSets" ]
+      do
+         deviceSet=$(oc get CephCluster ${ns} -o json | jq -r ".items[$clusterCount].spec.storage.storageClassDeviceSets[$deviceSetCount].name")
+         if [[ $deviceSet == $deviceSetName ]]
+         then
+            currentCount=$(oc get CephCluster ${ns} -o json | jq -r ".items[$clusterCount].spec.storage.storageClassDeviceSets[$deviceSetCount].count")
+            finalCount=$(( $currentCount + $count ))
+            echo "OSD count will be increased by ${count}.Total count for device set ${deviceSetName} is ${finalCount}."
+            if [ "${finalCount}" -gt "${maxCount}" ]
+            then
+               finalCount=${maxCount}
+               echo "DeviceSet ${deviceSet} capacity is full, cannot add more OSD to it"
+               break
+            fi
+            result=$(oc patch CephCluster rook-ceph ${ns} --type json --patch "[{ "op": "replace", "path": "/spec/storage/storageClassDeviceSets/${deviceSetCount}/count", "value": "${finalCount}" }]")
+            echo ${result}
+            break
+         fi
+         deviceSetCount=$((deviceSetCount+1))    
+      done
+      currentCount=$((currentCount+1))
+   done
+}
+
+function growOSD(){
+   i=0
+   alertmanagerroute=$(oc get route -n openshift-monitoring | grep alertmanager-main | awk '{ print $2 }')
+   curl -sk -H "Authorization: Bearer $(oc sa get-token prometheus-k8s -n openshift-monitoring)"  https://${alertmanagerroute}/api/v1/alerts | jq -r '.' >./tt.txt
+   export total_alerts=$(cat ./tt.txt | jq '.data | length')
```

Comment:
```suggestion
   export total_alerts=$(jq '.data | length' < tt.txt )
```

`cat`'s sole purpose is to concatenate files :)

Reply:
okay